modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/ReadWriteTransactionImpl.java

```diff
@@ -135,9 +140,24 @@ private void checkEnlistPossibility() {
         }
     }
 
-    /** {@inheritDoc} */
     @Override
-    protected CompletableFuture<Void> finish(boolean commit) {
+    public CompletableFuture<Void> commitAsync() {
+        return TransactionsExceptionMapperUtil.convertToPublicFuture(
+                finish(true, nullableHybridTimestamp(NULL_HYBRID_TIMESTAMP), false),
+                TX_COMMIT_ERR
+        );
+    }
+
+    @Override
+    public CompletableFuture<Void> rollbackAsync() {
+        return TransactionsExceptionMapperUtil.convertToPublicFuture(
+                finish(false, nullableHybridTimestamp(NULL_HYBRID_TIMESTAMP), false),
+                TX_ROLLBACK_ERR
+        );
+    }
+
+    @Override
+    public CompletableFuture<Void> finish(boolean commit, HybridTimestamp executionTimestamp, boolean full) {
```

Comment:
Why execution timestamp is not nullable?
This is to avoid constructing useless object.